### PR TITLE
Dev 03 - Added Tables for Restaurant/Category/Categorizations

### PR DESCRIPTION
**Pull request by Equable**
Fixed issues with rspec helper not using DatabaseCleaner(resolved factorybot issue)
Added Index/Show for restaurant. Show page is only a test route showing categories if present
Added tests for models and index page

### CODE REVIEW

File: app/assets/javascripts/categorizations.js [hunk 1]
```diff
@@ -0,0 +1,2 @@
+// Place all the behaviors and hooks related to the matching controller here.
+// All this logic will automatically be available in application.js.
```
Comment (nwalberts) [inline app/assets/javascripts/categorizations.js]: If there is no CSS in these files, then they are not needed. Additionally, you will likely be organizing your stylesheets not by model, but by feature or page.

File: app/controllers/restaurants_controller.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class RestaurantsController < ApplicationController
+
+	def index
+		@restaurants = Restaurant.all
+	end
+
+	def show 
+		@restaurant = Restaurant.find(params[:id])
+	end
+end
```
Comment (nwalberts) [inline app/controllers/restaurants_controller.rb:10]: This may just be github, but the indentation is a littly wonky here

File: app/helpers/restaurant_helper.rb [hunk 3]
```diff
@@ -0,0 +1,2 @@
+module RestaurantHelper
+end
```
Comment (nwalberts) [inline app/helpers/restaurant_helper.rb]: Why are these here? Please do not use generators or scaffolds. We dont want these empty bloat files

File: app/helpers/category_helper.rb [hunk 4]
```diff
@@ -0,0 +1,2 @@
+module CategoryHelper
+end
```
Comment (nwalberts) [inline app/helpers/category_helper.rb]: remove

File: app/helpers/categorizations_helper.rb [hunk 5]
```diff
@@ -0,0 +1,2 @@
+module CategorizationsHelper
+end
```
Comment (nwalberts) [inline app/helpers/categorizations_helper.rb]: remove

File: app/models/categorization.rb [hunk 6]
```diff
@@ -0,0 +1,6 @@
+class Categorization < ApplicationRecord
+	belongs_to :restaurant
+	belongs_to :category
+	validates :restaurant_id, presence: true
+	validates :category_id, presence: true
+end
```
Comment (nwalberts) [inline app/models/categorization.rb:4]: 1) strange indentation

2) In rails 5, you do not need these validations. They are automatic with a `belongs_to` relationship. Please remove

File: app/views/restaurants/show.html.erb [hunk 7]
```diff
@@ -0,0 +1,8 @@
+<p>This is the Show page for <%=@restaurant.name%> with test data to see if it properly connects to categories </p>
```
Comment (nwalberts) [inline app/views/restaurants/show.html.erb]: Pretend this code is ready for clients. You shouldnt be saying that there is test data in this app

File: db/seeds.rb [hunk 8]
```diff
@@ -3,5 +3,47 @@
...
+  	{ name: 'American' },
+  	{ name: 'French' },
+  	{ name: 'Pho' },
+  	{ name: 'Mediterranean' },
+  	{ name: 'Subs' }
+]
+
+categories.each do |x|
+	Category.create(x)
+end
```
Comment (nwalberts) [inline db/seeds.rb:22]: I know Im being a stickler, but please try to use representative variable names. Replace `x` with something like `category`

File: db/seeds.rb [hunk 9]
```diff
@@ -3,5 +3,47 @@
...
+	Restaurant.create(x)
+end
+
+categorizations = [
+	{restaurant: Restaurant.first, category: Category.fifth}
+]
+
+categorizations.each do |x|
+	Categorization.create(x)
+end
```
Comment (nwalberts) [inline db/seeds.rb:42]: use more representative variable names

File: spec/controllers/restaurants_controller_spec.rb [hunk 10]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe RestaurantsController, type: :controller do
+
+end
```
Comment (nwalberts) [inline spec/controllers/restaurants_controller_spec.rb]: delete this file, not being used

File: spec/features/restaurants/index_page_spec.rb [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+
+feature "Going to Restaurant Index page" do
+    let!(:restaurant1) { FactoryBot.create(:restaurant) }
+    let!(:restaurant2) { FactoryBot.create(:restaurant) }
+
+    scenario "should show a unorder list of restaurants with links and the restaurant names" do
+        visit '/restaurants'
+        
+        find('li a', text: restaurant1.name)
+        find('li a', text: restaurant2.name)
```
Comment (nwalberts) [inline spec/features/restaurants/index_page_spec.rb]: please use an expect statement rather than a query selector here.

File: spec/models/categories_spec.rb [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+			expect(category1.save).to eq(true)
+			expect(category2.save).to eq(false)
+		end
+
+		it "should store a unique name" do
+			expect(category3.save).to eq(true)
+		end
+	end
+
+end
```
Comment (nwalberts) [inline spec/models/categories_spec.rb:27]: We should have ActiveRecord `shoulda` validations in here. Refer to the Week 5 reading.

File: spec/models/categorizations_spec.rb [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+		end		
+
+	  it "category should have many restaurants through categoriations" do
+  		expect(Category.find(category.id).restaurants.first).to eq(restaurant)
+		end		
+
+		it "restaurant should have many categories through categorizations" do
+			expect(Restaurant.find(restaurant.id).categories.first).to eq(category)
+		end
+	end
```
Comment (nwalberts) [inline spec/models/categorizations_spec.rb:28]: Hmmm not bad testing these, but they should ideally be done with `shoulda` matchers from Week 5 testing

File: spec/models/restaurants_spec.rb [hunk 14]
```diff
@@ -0,0 +1,67 @@
...
+			expect(restaurant.save).to eq(false)
+		end
+
+		restaurant.city = 'hello'
+		restaurant.zip = nil
+		it "should not store value if zip is not specified" do 
+			expect(restaurant.save).to eq(false)
+		end
+	end
+end
```
Comment (nwalberts) [inline spec/models/restaurants_spec.rb:67]: ideally `shoulda` tests here as well

**Review by nwalberts — CHANGES_REQUESTED** [review]
Please address these items before moving on.

File: app/controllers/restaurants_controller.rb (hunk 2 above)
Comment (jkc909) [reply app/controllers/restaurants_controller.rb:10]: Not seeing any way for us to fix this, looks the same on both of our editors

File: spec/models/categories_spec.rb (hunk 12 above)
Comment (jkc909) [reply spec/models/categories_spec.rb:27]: talked to Nick, we'll be sure to use Shoulda moving forward

File: spec/models/categorizations_spec.rb (hunk 13 above)
Comment (jkc909) [reply spec/models/categorizations_spec.rb:28]: talked to Nick, we'll be sure to use Shoulda moving forward

File: spec/models/restaurants_spec.rb (hunk 14 above)
Comment (jkc909) [reply spec/models/restaurants_spec.rb:67]: talked to Nick, we'll be sure to use Shoulda moving forward

**Review by nwalberts — APPROVED** [review]
